11_gan.py

```diff
@@ -0,0 +1,76 @@
+import tensorflow as tf
+import numpy as np
+import matplotlib.pyplot as plt
+from tensorflow.examples.tutorials.mnist import input_data
+
+mnist = input_data.read_data_sets("MNIST_data/")
+images = mnist.train.images
+
+def xavier_initializer(shape):
+    return tf.random_normal(shape=shape, stddev=1/shape[0])
+
+# Generator
+z_size = 100
+g_w1_size = 400 
+g_out_size = 28 * 28
+
+# Discriminator
+x_size = 28 * 28
+d_w1_size = 400
+d_out_size = 1
+
+z = tf.placeholder('float', shape=(None, z_size))
+X = tf.placeholder('float', shape=(None, x_size))
+
+g_weights = {
+    'w1': tf.Variable(xavier_initializer(shape=(z_size, g_w1_size))),
+    'b1': tf.Variable(tf.zeros(shape=[g_w1_size])),
+    'out': tf.Variable(xavier_initializer(shape=(g_w1_size, g_out_size))),
+    'b2': tf.Variable(tf.zeros(shape=[g_out_size])),
+}
+
+d_weights ={
+    'w1': tf.Variable(xavier_initializer(shape=(x_size, d_w1_size))),
+    'b1': tf.Variable(tf.zeros(shape=[d_w1_size])),
+    'out': tf.Variable(xavier_initializer(shape=(d_w1_size, d_out_size))),
+    'b2': tf.Variable(tf.zeros(shape=[d_out_size])),
+}
+
+def G(z, w=g_weights):
```

Comment:
Do you think I should change this? I used this pattern because I didn't think extensibility was a concern, and I thought it makes the code more readable for two reasons:

- It makes the objective functions cleaner as I don't have to pass arguments ([lines 53-54](https://github.com/liviu-/TensorFlow-Tutorials/blob/b3f937e1b73bd26f2d9905eca5f3afd3c7bab96d/11_gan.py#L53-L54)) -- looks closer to the mathematical function from the paper.
- I could've just not passed the weights to the function, but I thought it makes it clearer that `G` is a function of input and weights, like you would specify mathematically.